src/psyclone/psyir/backend/fortran.py

```diff
@@ -572,3 +562,17 @@ def nemoimplicitloop_node(self, node):
 
         '''
         return "{0}{1}\n".format(self._nindent, str(node.ast))
+
+    @property
+    def directive_start(self):
+        ''':returns: "!${0}" - the start of a directive in Fortran
+        :rtype: str
+        '''
+        return "!${0}\n"
+
+    @property
+    def directive_end(self):
+        ''':returns: "!${0}" - the end of a directive in Fortran
```

Comment:
Probably lost in translation but obviously "!$" is not the end of a directive. Maybe using "closing directive" instead?

Reply:
Done.